Quarkus Maven Plugin fails on Maven 4

### Describe the bug

`mvn package` using latest Maven 4 alpha fails:
```
[INFO] --- quarkus:3.6.1:generate-code-tests (default) @ command-mode-quickstart ---
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  39.536 s (Wall Clock)
[INFO] Finished at: 2023-12-08T15:10:10-06:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.6.1:generate-code-tests (default) on project command-mode-quickstart: Quarkus code generation phase has failed: No valid constructor found on class org.apache.maven.internal.impl.DefaultTransport -> [Help 1]
```

### Expected behavior

Maven build succeeds

### Actual behavior

Maven build fails

### How to Reproduce?

- Download Maven 4 alpha, or use sdkman
- Create a Quarkus sample app (any)
- Try to build

### Output of `uname -a` or `ver`

Any

### Output of `java -version`

21.0.1

### Quarkus version or git rev

3.6.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 4.0.0-alpha-8

### Additional information

_No response_

Not necessarily a quarkus problem but I get the following warnings on `smallrye` dependencies building with RC2
```
[WARNING] 16 problems were encountered while building the effective model for io.smallrye.config:smallrye-config:jar:3.11.1 during dependency collection step for plugin (use -X to see details)
...
many others
```

Then, but not linked to the above, the build fails with
```
Quarkus code generation phase has failed: No matching bean available: type is interface org.eclipse.aether.RepositorySystem 

    Caused by: io.smallrye.beanbag.BeanInstantiationException: Failed to inject argument 4 of constructor for class org.eclipse.aether.internal.impl.DefaultRepositorySystem
        at io.smallrye.beanbag.ConstructorSupplier.get(ConstructorSupplier.java:30)
        at io.smallrye.beanbag.Bean$Pending.get(Bean.java:125)
        at io.smallrye.beanbag.Bean.get(Bean.java:47)
        at io.smallrye.beanbag.Scope.getBean(Scope.java:237)
        at io.smallrye.beanbag.Scope.requireBean(Scope.java:164)
        at io.smallrye.beanbag.BeanBag.requireBean(BeanBag.java:85)
        at io.smallrye.beanbag.maven.MavenFactory.getRepositorySystem(MavenFactory.java:167)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:923)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:297)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
        at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver(QuarkusWorkspaceProvider.java:115)

```

I tried this on my current project (using Quarkus 3.18.2 with java 21), would need to test with project generated from scratch.

I guess all of this is normal since Quarkus is explicitly depending on maven 3.9.9 (maven-core.version)
This will need work in [smallrye-beanbag](https://github.com/smallrye/smallrye-beanbag)

Still failing:
```
Apache Maven 4.0.0-rc-4-SNAPSHOT (f4c3fdd271caa33a80ccb0af776a5a4136b750bb)
Maven home: /Users/lprimak/Applications/maven/apache-maven-4.0.0-rc-4-20250523.151135
Java version: 24.0.1, vendor: Azul Systems, Inc., runtime: /Users/lprimak/.sdkman/candidates/java/24.0.1.fx-zulu/zulu-24.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "15.5", arch: "aarch64", family: "mac"
```
Stack trace:
```
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  2.941 s
[INFO] Finished at: 2025-06-03T09:09:32-05:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.23.0:generate-code-tests (default) on project quarkus-greeter: Quarkus code generation phase has failed: No matching bean available: type is interface org.eclipse.aether.RepositorySystem -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.23.0:generate-code-tests (default) on project quarkus-greeter: Quarkus code generation phase has failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:346)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
    at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
    at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
    at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
    at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
    at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke(Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
Caused by: org.apache.maven.plugin.MojoExecutionException: Quarkus code generation phase has failed
    at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:98)
    at io.quarkus.maven.GenerateCodeTestsMojo.doExecute(GenerateCodeTestsMojo.java:41)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:146)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:339)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
    at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
    at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
    at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
    at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
    at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke(Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
Caused by: io.smallrye.beanbag.NoSuchBeanException: No matching bean available: type is interface org.eclipse.aether.RepositorySystem
    at io.smallrye.beanbag.Scope.getBean(Scope.java:259)
    at io.smallrye.beanbag.Scope.requireBean(Scope.java:164)
    at io.smallrye.beanbag.BeanBag.requireBean(BeanBag.java:85)
    at io.smallrye.beanbag.maven.MavenFactory.getRepositorySystem(MavenFactory.java:167)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:934)
    at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:297)
    at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
    at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver(QuarkusWorkspaceProvider.java:115)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.artifactResolver(QuarkusBootstrapProvider.java:236)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap(QuarkusBootstrapProvider.java:259)
    at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication(QuarkusBootstrapProvider.java:411)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:157)
    at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:152)
    at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication(QuarkusBootstrapMojo.java:307)
    at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:83)
    at io.quarkus.maven.GenerateCodeTestsMojo.doExecute(GenerateCodeTestsMojo.java:41)
    at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:182)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:146)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:339)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
    at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
    at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
    at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
    at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
    at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
    at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
    at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
    at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke(Method.java:565)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
    Suppressed: io.smallrye.beanbag.BeanInstantiationException: Failed to inject argument 4 of constructor for class org.eclipse.aether.internal.impl.DefaultRepositorySystem
        at io.smallrye.beanbag.Bean$Failed.get(Bean.java:82)
        at io.smallrye.beanbag.Bean$Pending.get(Bean.java:129)
        at io.smallrye.beanbag.Bean.get(Bean.java:47)
        at io.smallrye.beanbag.Scope.getBean(Scope.java:237)
        at io.smallrye.beanbag.Scope.requireBean(Scope.java:164)
        at io.smallrye.beanbag.BeanBag.requireBean(BeanBag.java:85)
        at io.smallrye.beanbag.maven.MavenFactory.getRepositorySystem(MavenFactory.java:167)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:934)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:297)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
        at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver(QuarkusWorkspaceProvider.java:115)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.artifactResolver(QuarkusBootstrapProvider.java:236)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap(QuarkusBootstrapProvider.java:259)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication(QuarkusBootstrapProvider.java:411)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:157)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:152)
        at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication(QuarkusBootstrapMojo.java:307)
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:83)
        at io.quarkus.maven.GenerateCodeTestsMojo.doExecute(GenerateCodeTestsMojo.java:41)
        at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:182)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:146)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:339)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
        at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
        at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
        at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
        at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
        at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.lang.reflect.Method.invoke(Method.java:565)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
    Caused by: io.smallrye.beanbag.BeanInstantiationException: Failed to inject argument 4 of constructor for class org.eclipse.aether.internal.impl.DefaultRepositorySystem
        at io.smallrye.beanbag.ConstructorSupplier.get(ConstructorSupplier.java:30)
        at io.smallrye.beanbag.Bean$Pending.get(Bean.java:125)
        at io.smallrye.beanbag.Bean.get(Bean.java:47)
        at io.smallrye.beanbag.Scope.getBean(Scope.java:237)
        at io.smallrye.beanbag.Scope.requireBean(Scope.java:164)
        at io.smallrye.beanbag.BeanBag.requireBean(BeanBag.java:85)
        at io.smallrye.beanbag.maven.MavenFactory.getRepositorySystem(MavenFactory.java:167)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:934)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:297)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
        at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver(QuarkusWorkspaceProvider.java:115)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.artifactResolver(QuarkusBootstrapProvider.java:236)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap(QuarkusBootstrapProvider.java:259)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication(QuarkusBootstrapProvider.java:411)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:157)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:152)
        at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication(QuarkusBootstrapMojo.java:307)
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:83)
        at io.quarkus.maven.GenerateCodeTestsMojo.doExecute(GenerateCodeTestsMojo.java:41)
        at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:182)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:146)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:339)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
        at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
        at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
        at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
        at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
        at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.lang.reflect.Method.invoke(Method.java:565)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
    Caused by: io.smallrye.beanbag.NoSuchBeanException: No matching bean available: type is interface org.eclipse.aether.impl.ArtifactDescriptorReader
        at io.smallrye.beanbag.Scope.getBean(Scope.java:259)
        at io.smallrye.beanbag.BeanResolver.get(BeanResolver.java:20)
        at io.smallrye.beanbag.ConstructorSupplier.get(ConstructorSupplier.java:27)
        at io.smallrye.beanbag.Bean$Pending.get(Bean.java:125)
        at io.smallrye.beanbag.Bean.get(Bean.java:47)
        at io.smallrye.beanbag.Scope.getBean(Scope.java:237)
        at io.smallrye.beanbag.Scope.requireBean(Scope.java:164)
        at io.smallrye.beanbag.BeanBag.requireBean(BeanBag.java:85)
        at io.smallrye.beanbag.maven.MavenFactory.getRepositorySystem(MavenFactory.java:167)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.initRepoSystemAndManager(BootstrapMavenContext.java:934)
        at io.quarkus.bootstrap.resolver.maven.BootstrapMavenContext.getRepositorySystem(BootstrapMavenContext.java:297)
        at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.<init>(MavenArtifactResolver.java:114)
        at io.quarkus.maven.components.QuarkusWorkspaceProvider.createArtifactResolver(QuarkusWorkspaceProvider.java:115)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.artifactResolver(QuarkusBootstrapProvider.java:236)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.doBootstrap(QuarkusBootstrapProvider.java:259)
        at io.quarkus.maven.QuarkusBootstrapProvider$QuarkusMavenAppBootstrap.bootstrapApplication(QuarkusBootstrapProvider.java:411)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:157)
        at io.quarkus.maven.QuarkusBootstrapProvider.bootstrapApplication(QuarkusBootstrapProvider.java:152)
        at io.quarkus.maven.QuarkusBootstrapMojo.bootstrapApplication(QuarkusBootstrapMojo.java:307)
        at io.quarkus.maven.GenerateCodeMojo.generateCode(GenerateCodeMojo.java:83)
        at io.quarkus.maven.GenerateCodeTestsMojo.doExecute(GenerateCodeTestsMojo.java:41)
        at io.quarkus.maven.QuarkusBootstrapMojo.execute(QuarkusBootstrapMojo.java:182)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:146)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:339)
        at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:310)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:214)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:179)
        at org.apache.maven.lifecycle.internal.MojoExecutor$1.run(MojoExecutor.java:168)
        at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute(DefaultMojosExecutionStrategy.java:39)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:165)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:110)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:76)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:60)
        at org.apache.maven.lifecycle.internal.DefaultLifecycleStarter.execute(DefaultLifecycleStarter.java:123)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:310)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:225)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:149)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.doExecute(MavenInvoker.java:461)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:100)
        at org.apache.maven.cling.invoker.mvn.MavenInvoker.execute(MavenInvoker.java:81)
        at org.apache.maven.cling.invoker.LookupInvoker.doInvoke(LookupInvoker.java:166)
        at org.apache.maven.cling.invoker.LookupInvoker.invoke(LookupInvoker.java:136)
        at org.apache.maven.cling.ClingSupport.run(ClingSupport.java:76)
        at org.apache.maven.cling.MavenCling.main(MavenCling.java:51)
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.lang.reflect.Method.invoke(Method.java:565)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:255)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:201)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
[ERROR] 
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn [args] -r
```

--- a/independent-projects/bootstrap/maven-resolver/pom.xml
+++ b/independent-projects/bootstrap/maven-resolver/pom.xml
@@ -76,6 +76,17 @@
             <artifactId>slf4j-api</artifactId>
         </dependency>
         <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-bootstrap-maven4-resolver</artifactId>
+            <version>${project.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
             <groupId>org.apache.maven</groupId>
             <artifactId>maven-embedder</artifactId>
         </dependency>

--- a/independent-projects/bootstrap/maven4-resolver/pom.xml
+++ b/independent-projects/bootstrap/maven4-resolver/pom.xml
@@ -1,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-bootstrap-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-bootstrap-maven4-resolver</artifactId>
+    <name>Quarkus - Bootstrap - Maven 4 Resolver</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <version>1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.sisu</groupId>
+            <artifactId>org.eclipse.sisu.inject</artifactId>
+            <version>0.9.0.M3</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-resolver-provider</artifactId>
+            <version>4.0.0-rc-3</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>true</filtering>
+            </resource>
+        </resources>
+    </build>
+</project>

--- a/independent-projects/bootstrap/maven4-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapArtifactTransformer.java
+++ b/independent-projects/bootstrap/maven4-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/BootstrapArtifactTransformer.java
@@ -1,0 +1,13 @@
+package io.quarkus.bootstrap.resolver.maven;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.eclipse.aether.spi.artifact.transformer.ArtifactTransformer;
+import org.eclipse.sisu.Priority;
+
+@Singleton
+@Named
+@Priority(100)
+public class BootstrapArtifactTransformer implements ArtifactTransformer {
+}

--- a/independent-projects/bootstrap/maven4-resolver/src/main/resources/META-INF/sisu/javax.inject.Named
+++ b/independent-projects/bootstrap/maven4-resolver/src/main/resources/META-INF/sisu/javax.inject.Named
@@ -1,0 +1,13 @@
+org.apache.maven.repository.internal.DefaultArtifactDescriptorReader
+org.apache.maven.repository.internal.DefaultModelCacheFactory
+org.apache.maven.repository.internal.DefaultModelVersionParser
+org.apache.maven.repository.internal.DefaultVersionRangeResolver
+org.apache.maven.repository.internal.DefaultVersionResolver
+org.apache.maven.repository.internal.DefaultVersionSchemeProvider
+org.apache.maven.repository.internal.PluginsMetadataGeneratorFactory
+org.apache.maven.repository.internal.SnapshotMetadataGeneratorFactory
+org.apache.maven.repository.internal.VersionsMetadataGeneratorFactory
+org.apache.maven.repository.internal.relocation.DistributionManagementArtifactRelocationSource
+org.apache.maven.repository.internal.relocation.UserPropertiesArtifactRelocationSource
+org.apache.maven.repository.internal.type.DefaultTypeProvider
+io.quarkus.bootstrap.resolver.maven.BootstrapArtifactTransformer

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -79,6 +79,7 @@
         <module>bom</module>
         <module>bom-test</module>
         <module>app-model</module>
+        <module>maven4-resolver</module>
         <module>maven-resolver</module>
         <module>core</module>
         <module>runner</module>

